frontend/src/views/DashboardPage/DashboardPage.tsx

```diff
@@ -746,8 +809,190 @@ export const DashboardPage = () => {
     ({ isReadDenied, isWriteDenied }) => !isReadDenied || !isWriteDenied
   );
 
+
+  const handleCheckedSecret = (secretObj: { _id: string, isChecked: string | boolean }) => {
+    const checkedSecretsClone = [...checkedSecrets]
+    const checkedSecretIndex = checkedSecretsClone.findIndex(secret => secret._id === secretObj._id)
+    if (secretObj.isChecked) {
+      checkedSecretsClone.push(secretObj)
+    } else {
+      checkedSecretsClone.splice(checkedSecretIndex, 1)
+    }
+    setCheckedSecrets(() => checkedSecretsClone)
+  }
+
+  const onMoveSecrets = async ($folderId: string, $checkedSecrets: { _id: string, isChecked: string | boolean }[]) => {
```

Comment:
Instead of alert use the delete action modal to ask user's for confirmation. This will allow uniform ui